server/partition.go

```diff
@@ -223,6 +226,27 @@ func (p *partition) IsPaused() bool {
 	return p.paused
 }
 
+// SetReadonly sets the partition's readonly flag.
+func (p *partition) SetReadonly(readonly bool) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.readonly = readonly
+
+	// Notify that the partition is now readonly.
+	if readonly && !p.readonly {
```

Comment:
Shouldn't this check occur _before_ `p.readonly = readonly`? Unless I'm reading this wrong...

Reply:
Fixed in f93da8e.